### PR TITLE
Jobservice helm

### LINKED ISSUE #1174: Helm chart for the job service
We should create a helm chart for the job service so we can deploy into kubernetes.

### PR DESCRIPTION
**Pull request by kannon92**
Closes #1174 

We created a deployment folder in jobservice and this includes the necessary items for deployment.  

1) StatefulSet
2) Service
3) Secret
4) ServiceAccount (Told we need this)

Also had to make changes in the code to be able to read configs from different locations.

### CODE REVIEW

File: deployment/jobservice/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+kind: ServiceAccount
+metadata:
+  name: {{ include "jobservice.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "jobservice.labels.all" . | nindent 4 }}
+  {{ if .Values.serviceAccount }}
+  {{ toYaml .Values.serviceAccount }}
+  {{ end }}
+  {{ end }}
```
Comment (JamesMurkin) [inline deployment/jobservice/templates/serviceaccount.yaml:12]: Should the second {{ end }} be at the start of the line?

File: deployment/jobservice/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    - key: app
+                      operator: In
+                      values:
+                        - {{ include "jobservice.name" . }}
+                topologyKey: kubernetes.io/hostname
+      volumes:
+        - name: user-config
+          secret:
+            secretName: {{ include "jobservice.config.name" . }}
+        {{- if .Values.additionalVolumes }}
```
Comment (JamesMurkin) [inline deployment/jobservice/templates/statefulset.yaml:70]: It'd be neat if we could support some standard pvc/emptydir setup. However maybe this is the standard approach for statefulsets publically.

Just a thought and nothing major

File: deployment/jobservice/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "jobservice.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "jobservice.labels.all" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicas }}
```
Comment (JamesMurkin) [inline deployment/jobservice/templates/statefulset.yaml:9]: If we only support 1 replica, should replicas be configurable?

File: deployment/jobservice/templates/serviceaccount.yaml (hunk 1 above)
Comment (kannon92) [reply deployment/jobservice/templates/serviceaccount.yaml]: Yep!  Thanks for that.

File: deployment/jobservice/templates/statefulset.yaml (hunk 3 above)
Comment (kannon92) [reply deployment/jobservice/templates/statefulset.yaml]: Done.

File: deployment/jobservice/templates/statefulset.yaml (hunk 2 above)
Comment (kannon92) [reply deployment/jobservice/templates/statefulset.yaml:70]: We discussed and we will hold on this.  I'm unclear of how the PVC stuff will go with internal k8 so I'd like to hold on it.

**Review by JamesMurkin — APPROVED** [review]

**Pull request merged**